test_scripts/API/Alert/007_ATF_Alert_UI_Alert_WARNINGS.lua

```diff
@@ -0,0 +1,162 @@
+
+--[[ General Settings for configuration ]]
+Test = require('connecttest')
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:Precondition_ActivationApp()
+  local request_id = self.hmiConnection:SendRequest("SDL.ActivateApp", { appID = self.applications[config.application1.registerAppInterfaceParams.appName]})
+  EXPECT_HMIRESPONSE(request_id)
+  :Do(function(_,data)
+    if (data.result.isSDLAllowed ~= true) then
+      local request_id1 = self.hmiConnection:SendRequest("SDL.GetUserFriendlyMessage", {language = "EN-US", messageCodes = {"DataConsent"}})
+      EXPECT_HMIRESPONSE(request_id1)
+      :Do(function(_,_)
+        self.hmiConnection:SendNotification("SDL.OnAllowSDLFunctionality", {allowed = true, source = "GUI", device = {id = config.deviceMAC, name = ServerAddress}})
+        EXPECT_HMICALL("BasicCommunication.ActivateApp")
+        :Do(function(_,data1)
+          self.hmiConnection:SendResponse(data1.id,"BasicCommunication.ActivateApp", "SUCCESS", {})
+        end)
+      end)
+    end
+  end)
+  EXPECT_NOTIFICATION("OnHMIStatus", {hmiLevel = "FULL", systemContext = "MAIN"})
+end
+
+commonSteps:PutFile("Precondition_PutFile", "icon.png")
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+local resultCodes = {"SUCCESS", "WARNINGS", "WRONG_LANGUAGE", "RETRY", "SAVED"}
+for i=1,#resultCodes do
+  Test["TestStep_Alert_UI_Alert_WARNINGS_TTS_Speak_"..resultCodes[i]] = function(self)
+    local cor_id = self.mobileSession:SendRPC("Alert",
+    {
+      alertText1 = "alertText1",
+      alertText2 = "alertText2",
+      alertText3 = "alertText3",
+      ttsChunks = {{text = "TTSChunk", type = "TEXT"}},
+      duration = 3000,
+      progressIndicator = true,
+      softButtons =
+        {{
+          type = "BOTH",
+          text = "Close",
+          image = { value = "icon.png", imageType = "DYNAMIC"},
+          isHighlighted = true,
+          softButtonID = 3,
+          systemAction = "DEFAULT_ACTION",
+      }}
+    })
+
+    EXPECT_HMICALL("UI.Alert",
+    {alertStrings =
+      {
+        {fieldName = "alertText1", fieldText = "alertText1"},
+        {fieldName = "alertText2", fieldText = "alertText2"},
+        {fieldName = "alertText3", fieldText = "alertText3"}
+      },
+      alertType = "BOTH",
+      duration = 3000,
+      progressIndicator = true,
+      softButtons =
+        {{
+          type = "BOTH",
+          text = "Close",
+          image = { imageType = "DYNAMIC"},
+          isHighlighted = true,
+          softButtonID = 3,
+          systemAction = "DEFAULT_ACTION",
+      }},
+      appID = self.applications[config.application1.registerAppInterfaceParams.appName]
+    })
+    :ValidIf(function(_,data)
+      local value_Icon = storagePath .. "icon.png"
+      if (string.match(data.params.cmdIcon.value, "%S*" .. "("..string.sub(storagePath, 2).."icon.png)" .. "$") == nil ) then
```

Comment:
@vvvakulenko, image is part of array softButtons, so verification should be done with data.params.softButtons[1].image.value, instead of data.params.cmdIcon.value